### PR TITLE
feat(include_y): found_male --> include_y

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- Removes the counters: 'found_female', 'found_male', 'found_other' from active_parameters and replaces it with the bolean 'include_y'

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Active_parameter.pm [hunk 1]
```diff
@@ -1783,15 +1784,60 @@ sub set_gender_sample_ids {
...
+
+    $active_parameter_href->{include_y} = 0;
+
+  GENDER:
+    foreach my $gender ( keys %{ $active_parameter_href->{gender} } ) {
+
+        next GENDER if ( $gender eq q{females} );
+
+        $active_parameter_href->{include_y} =
+          ( @{ $active_parameter_href->{gender}{$gender} } > 0 ) ? 1 : 0;
```
Comment (henrikstranneheim) [inline lib/MIP/Active_parameter.pm:1837]: Ternary operator - how nice! We should use it a bit more than we have done before
```suggestion
          @{ $active_parameter_href->{gender}{$gender} } > 0 ? 1 : 0;
```
Do you need the ()?

Comment (henrikstranneheim) [reply lib/MIP/Active_parameter.pm:1837]: Actually do you need the "> 0" as well?

File: lib/MIP/Parse/Gender.pm [hunk 2]
```diff
@@ -483,25 +484,39 @@ sub update_gender_info {
...
+
+        $active_parameter_href->{gender_estimation}{$sample_id} = q{female};
+    }
 
-    ## Decrement found_male
-    $active_parameter_href->{found_male}--;
-    $active_parameter_href->{gender_estimation}{$sample_id} = q{female};
+    ## Remove sample_id from others
+    @{ $active_parameter_href->{gender}{others} } =
+      grep { !/$sample_id/xms } @{ $active_parameter_href->{gender}{others} };
```
Comment (henrikstranneheim) [inline lib/MIP/Parse/Gender.pm:505]:
```suggestion
      grep { not /$sample_id/xms } @{ $active_parameter_href->{gender}{others} };
```

File: t/set_gender_sample_ids.t [hunk 3]
```diff
@@ -68,45 +68,29 @@ my %sample_info = (
...
+is_deeply( $active_parameter{gender},
+    \%expected_gender_info, q{Added gender info to active parameter} );
+is( $active_parameter{include_y}, 1, q{Set_include y} );
 
 ## Given no males or females
 # Clear data from previous test
-map { $active_parameter{$_} = 0 } (qw{ found_male found_female found_other });
+delete $active_parameter{gender};
+
+#map { $active_parameter{$_} = 0 } (qw{ found_male found_female found_other });
```
Comment (henrikstranneheim) [inline t/set_gender_sample_ids.t:93]:
```suggestion
```

File: t/update_gender_info.t [hunk 4]
```diff
@@ -89,17 +90,20 @@ my $is_ok = update_gender_info(
...
 ok( $is_ok, q{Updated gender info} );
 
-## Then increment found male and add estimated gender for sample id
-is( $active_parameter{found_male}, 1, q{Incremented found male} );
+## Then set include_y to 1 and add estimated gender for sample id
+is( $active_parameter{include_y}, 1, q{Include y} );
 is( $active_parameter{gender_estimation}{$sample_id},
     q{male}, q{Added estimated gender male} );
+is_deeply( $active_parameter{gender}{males}, [$sample_id], q{Add to males sample_id} );
+is_deeply( $active_parameter{gender}{others}, [], q{Remove from others sample_id} );
```
Comment (henrikstranneheim) [inline t/update_gender_info.t:101]:
```suggestion
is( @{ $active_parameter{gender}{others} }, 0, q{Remove from others sample_id} );
```

File: t/update_gender_info.t [hunk 5]
```diff
@@ -111,9 +115,12 @@ update_gender_info(
...
 
-## Then increment found female and add estimated gender for sample id
-is( $active_parameter{found_male}, 0, q{Decremented found male} );
+## Then set include y to zero and add estimated gender for sample id
+is( $active_parameter{include_y}, 0, q{Exclude y} );
 is( $active_parameter{gender_estimation}{$sample_id},
     q{female}, q{Added estimated gender female} );
+is_deeply( $active_parameter{gender}{females}, [$sample_id],
+    q{Add to females sample_id} );
+is_deeply( $active_parameter{gender}{others}, [], q{Remove from others sample_id} );
```
Comment (henrikstranneheim) [inline t/update_gender_info.t:124]:
```suggestion
is( @{ $active_parameter{gender}{others} }, 0, q{Remove from others sample_id} );
```

File: lib/MIP/Active_parameter.pm (hunk 1 above)
Comment (jemten) [reply lib/MIP/Active_parameter.pm:1837]: You're right! The '> 0' is unnecessary

**Pull request merged**